### PR TITLE
feat(ci): transition release pipeline to automated semver and refresh branding

### PR DESCRIPTION
**Pull request by fbraz3**
## Context
This PR transitions GeneralsX from the legacy `Beta-XX` release workflow to an automated Semantic Versioning (SemVer) release pipeline in preparation for the milestone 1.0 stable release, fixes macOS distribution packaging, adopts a cleaner military insignia visual identity, and refines release documentation.

> [!NOTE]
> This PR is intended to be merged **after NGMP PR** #260.

---

### 1. Automated Semantic Versioning (SemVer)
- **Base Versioning Anchor**: Introduced `base-version.txt` (initialized to `1.0`) at repo root to define the current major/minor release stream.
- **Dynamic Version Calculation**: Added `scripts/tooling/release/calculate_next_version.sh` to automatically detect highest existing tags matching `MAJOR.MINOR.*` and calculate the next patch level (`1.0.0`, `1.0.1`, etc.).
- **Automated Test Suite**: Added `scripts/qa/test_calculate_next_version.sh` with 9 passing unit test cases covering edge cases, numeric sorting, and empty repositories.
- **CI/CD Pipeline Update**: Updated `.github/workflows/release.yml` to remove manual version prompting, enforce global non-overlapping concurrency, support Dry Run without tag creation, and search changelogs across both SemVer and legacy beta tags.
- **Watermark Formatting**: Updated `Core/GameEngine/Source/Common/version.cpp` to format project watermark cleanly as `GeneralsX - Cross Platform C&C Generals - 1.0.0` while maintaining backward compatibility with legacy tags.

### 2. Native macOS Zip Packaging
- Fixed macOS distribution bundle packaging in `scripts/build/macos/bundle-macos-zh.sh` and `bundle-macos-generals.sh` using `zip -y -r` to preserve internal symlinks.
- Updated `.github/workflows/build-macos.yml`, `.github/workflows/replay-tests.yml`, and `.github/workflows/release.yml` to distribute clean `macOS-GeneralsXZH.zip` and `macOS-GeneralsX.zip` archives, resolving double-compressed `.tar.zip` extraction issues in macOS Finder.

### 3. Visual Identity & Windows Executable Icons
- Replaced app icons (`assets/generalsx_icon.png` and `assets/generalsx-zh_icon.png`) with classic military insignia emblems combined with the GeneralsX "X" motif for superior readability and authenticity.
- Created `scripts/tooling/generate_windows_icons.py` and regenerated multi-resolution Windows executable icons (`Generals/Code/Main/Generals.ico` and `GeneralsMD/Code/Main/Generals.ico`) across 7 resolutions (16x16 to 256x256, 32-bit RGBA).
- Removed obsolete concept asset `assets/generalsx_splash.png` (3.1 MB).

### 4. Release Template & Prompt Polishing
- Reordered release notes so `# Getting Started` precedes `# What's New`.
- Simplified the top-level issue disclaimer to clean standard text.
- Synchronized community ports list across `README.md`, `.github/workflows/release.yml`, and `.github/prompts/prepare-release-draft.prompt.md` (adding `wasm-generals` and removing inactive `GeneralsXWeb`).

---

## Validation
- [x] Unit test suite `scripts/qa/test_calculate_next_version.sh` passed 9/9 tests.
- [x] Validated YAML syntax for all workflows with `ruby -ryaml`.
- [x] Inspected multi-resolution `.ico` binaries (7 sub-images, RGBA).
- [x] Verified macOS symlink preservation with `zip -y`.
- [x] Updated diary in `docs/WORKLOG/2026-09-DIARY.md`.

## Summary by CodeRabbit

- **New Features**
  - Releases now automatically calculate Semantic Versions from the base version and existing tags.
  - Added Windows ZIP artifacts and multi-resolution Windows icons.
  - macOS releases are packaged as ZIP archives containing only the app bundle.
  - Linux Flatpak and cross-platform release artifacts now use consistent naming.

- **Bug Fixes**
  - Improved artifact downloading, extraction, executable permissions, and packaging reliability.
  - Version watermarks now support numeric Semantic Version and legacy tag formats.

- **Documentation**
  - Updated release pipeline guidance and release-note templates.
  - Removed the GeneralsXWeb community port from listed options.

### CODE REVIEW

File: docs/WORKLOG/2026-09-DIARY.md [hunk 1]
```diff
@@ -41,6 +41,41 @@
...
+### Remove Obsolete Splash Screen Asset
+- Removed `assets/generalsx_splash.png` (3.1MB) which had no code or pipeline references and was left over from older concept mockups.
+- Kept the asset folder focused exclusively on retail-aligned military insignia icons.
+
+### Update Authentication Portal Branding Logo (`login.generalsx.org`)
+- Updated authentication portal brand assets (`tmp/generals-login`) with the new high-resolution logo (`public/assets/images/logo3.png`).
+- Replaced references across all portal templates (`index.php`, `success.php`, `error.php`, `termsofservice.php`, `codeofconduct.php`, `privacypolicy.php`).
+- Validated PHP syntax across all templates.
+- Pushed commit `68c414a` to `fbraz3/generalsx-login` (`origin/master`).
+- Deployed live changes to `/data/docker/generalslogin` on production host `web04-ctb.braz.cloud`.
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply docs/WORKLOG/2026-09-DIARY.md]: Addressed in bb6507790. Removed the hardcoded filesystem path and replaced it with a generic deployment directory description.

File: scripts/qa/test_calculate_next_version.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+# Test 7: Whitespace trimming in base-version.txt
+echo -e "  2.0 \n " > "${REPO1}/base-version.txt"
+OUTPUT7="$(cd "$REPO1" && "$CALCULATE_SCRIPT")"
+assert_equals "Whitespace handling for 2.0" "2.0.0" "$OUTPUT7"
+
+# Test 8: GITHUB_OUTPUT writing
+GH_OUT_FILE="${TEST_DIR}/github_output.txt"
+touch "$GH_OUT_FILE"
+GITHUB_OUTPUT="$GH_OUT_FILE" "$CALCULATE_SCRIPT" "${REPO1}/base-version.txt" >/dev/null
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply scripts/qa/test_calculate_next_version.sh]: Addressed in bb6507790. Isolated GITHUB_OUTPUT test execution to run within the REPO1 temporary directory.

File: scripts/tooling/release/calculate_next_version.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+if [ -z "$RAW_BASE" ]; then
+  echo "ERROR: Version base file '$VERSION_FILE' is empty" >&2
+  exit 1
+fi
+
+# Extract MAJOR and MINOR components
+MAJOR="$(echo "$RAW_BASE" | cut -d'.' -f1)"
+MINOR="$(echo "$RAW_BASE" | cut -d'.' -f2)"
+
+if ! [[ "$MAJOR" =~ ^[0-9]+$ ]] || ! [[ "$MINOR" =~ ^[0-9]+$ ]]; then
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply scripts/tooling/release/calculate_next_version.sh]: Addressed in bb6507790. Hardened base-version validation with regex ^[0-9]+\.[0-9]+$ to strictly require two numeric components and reject 3-part versions.

File: .github/workflows/build-macos.yml [hunk 4]
```diff
@@ -480,6 +476,6 @@ jobs:
         uses: actions/upload-artifact@v7
         with:
           name: macos-${{ matrix.game.id }}-app
-          path: /tmp/macos-artifact-${{ matrix.game.id }}/*.tar
+          path: /tmp/macos-artifact-${{ matrix.game.id }}/${{ matrix.game.output }}.app
```
Comment (arazmj) [inline .github/workflows/build-macos.yml]: **High — upload the generated ZIP, not the extracted `.app` directory.** `upload-artifact` stores the contents of the path, so this artifact downloads as `Contents/...` without a `GeneralsX*.app` root. The current macOS replay job already fails because `replay-tests.yml` cannot find an `.app`, ZIP, or TAR. The release workflow makes the same assumption and will fail too. File modes are also normalized by artifact transport, so directly uploaded app contents lose executable permissions. The bundle scripts already produce a root-correct ZIP with `zip -y`, which preserves the app directory, executable bits, and symlinks; upload that ZIP as the artifact and let replay/release consumers extract it.

**Review by arazmj — CHANGES_REQUESTED** [review]
Two release-blocking issues remain.

1. **Medium — `.github/workflows/release.yml:81-117`: release-preparation failures leave an orphan tag.** `cleanup-tag-on-failure` only depends on and checks the three build jobs. If artifact normalization, changelog generation, or `gh release create` fails or is cancelled after all builds succeed, the pushed tag is never deleted. With automatic tag scanning, the next run then advances from (for example) `1.0.0` to `1.0.1` instead of retrying the failed release. Add cleanup that also observes `prepare-release` failure/cancellation (while verifying that a successful release was not created), or defer remote tag publication until preparation can complete.

2. The inline macOS artifact finding is demonstrated by the current failed replay job and also affects release packaging.

File: .github/workflows/build-macos.yml (hunk 4 above)
Comment (fbraz3) [reply .github/workflows/build-macos.yml]: Addressed both points in commit 1ea656600:

1. **macOS Artifact Packaging**: `build-macos.yml` now uploads the root-correct `.zip` archive produced by `bundle-macos-*.sh` (`zip -y -r`), preserving executable file permissions (`+x`), symlinks, and the single `.app` bundle directly at the root. `replay-tests.yml` unpacks this archive with `unzip -q`, and `release.yml` deploys this pre-built archive directly as the release asset (`macOS-GeneralsXZH.zip` / `macOS-GeneralsX.zip`).

2. **Orphan Tag Cleanup**: `cleanup-tag-on-failure` in `release.yml` now depends on all build jobs as well as `prepare-release` (`needs: [create-tag, build-linux-flatpak, build-macos, build-windows, prepare-release]`). It observes failure or cancellation of both builds and release preparation, and verifies via `gh release view` that a published release does not already exist before deleting the tag, preventing orphaned tags from advancing SemVer on subsequent runs.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -419,3 +429,46 @@ jobs:
...
+      - name: Delete orphaned tag
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          VERSION="${{ needs.create-tag.outputs.release_version }}"
+          # Verify that a release was not already created before deleting the tag
+          if gh release view "$VERSION" >/dev/null 2>&1; then
+            echo "Release $VERSION already exists on GitHub; skipping tag deletion."
+            exit 0
+          fi
```
Comment (arazmj) [inline .github/workflows/release.yml]: **[P2] Do not authorize tag deletion when the release lookup fails.** A nonzero exit from `gh release view` can mean a timeout, HTTP 5xx, or an authentication/rate-limit error, not just a missing release. If release creation succeeded remotely before the CLI failed or the job was cancelled, a failed lookup here falls through to `git push --delete` and removes the tag of an existing release. With a mocked HTTP-503 lookup, this exact script invokes the deletion and exits successfully. Query an endpoint whose HTTP status can be distinguished: preserve the tag when a release exists, delete only on a confirmed not-found response, and fail without mutating the tag on other errors. This is separate from the deletion-error suppression already reported below.

**Review by arazmj — CHANGES_REQUESTED** [review]
Re-reviewed 1ea656600 against the previously reviewed 228072a7a.

The earlier macOS packaging finding is resolved: the producer uploads the generated ZIP, replay extracts it, and release publication copies it unchanged. The current macOS replay job also passes. The cleanup job now correctly depends on prepare-release and handles failure/cancellation, resolving the earlier scheduling gap.

Two cleanup error-handling problems remain: the new release-existence query can authorize deletion on API failure (inline finding), and the existing CodeRabbit thread on lines 471-473 correctly identifies that failed remote tag deletion is reported as a successful no-op. I reproduced both paths using the exact cleanup script with mocked gh/git commands; no remote tags were modified. Please distinguish confirmed absence from lookup failure and propagate deletion failures.

File: .github/workflows/build-macos.yml (hunk 4 above)
Comment (arazmj) [reply .github/workflows/build-macos.yml]: Confirmed fixed in 1ea656600: the generated ZIP is uploaded intact, replay extraction preserves the bundle, and release preparation copies the original ZIP. Resolving this packaging thread; remaining cleanup error-handling concerns are recorded in the new review.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -419,3 +429,46 @@ jobs:
...
+
+          echo "Pipeline failed or cancelled before release completion. Deleting orphaned tag: $VERSION"
+          git push origin --delete "refs/tags/$VERSION" \
+            && echo "Tag $VERSION deleted from remote." \
+            || echo "Tag $VERSION not found on remote, nothing to delete."
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply .github/workflows/release.yml]: Addressed in d4c66a2dc. The cleanup step now queries `git ls-remote --exit-code --tags origin refs/tags/$VERSION` to distinguish status 2 (confirmed tag absence, safe no-op) from other query errors, and executes `git push origin --delete` directly so remote deletion failures propagate as job failures.

File: .github/workflows/release.yml (hunk 5 above)
Comment (fbraz3) [reply .github/workflows/release.yml]: Addressed in d4c66a2dc. The release lookup now queries `gh api --include repos/${{ github.repository }}/releases/tags/$VERSION` and parses the HTTP status code explicitly. Tag deletion is authorized strictly on confirmed HTTP 404. On HTTP 200, tag deletion is skipped (release verified as existing); on all other responses (5xx, 403, network timeouts), the step aborts with an error without mutating the remote tag.

File: .github/workflows/build-linux-flatpak.yml [hunk 7]
```diff
@@ -124,7 +124,8 @@ jobs:
         if: success()
         uses: actions/upload-artifact@v7
         with:
-          name: linux-flatpak-${{ matrix.game.id }}-${{ inputs.preset }}
+          name: ${{ matrix.game.output }}-${{ inputs.preset }}.flatpak
           path: ${{ steps.bundle.outputs.path }}
+          archive: false
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply .github/workflows/build-linux-flatpak.yml:134]: Addressed in d4c66a2dc. In `build-linux-flatpak.yml`, when the candidate bundle located via the fallback differs from `${EXPECTED}`, it is moved to `${EXPECTED}` before exporting `steps.bundle.outputs.path`, guaranteeing canonical naming under `archive: false` uploads.

File: .github/workflows/build-windows.yml [hunk 8]
```diff
@@ -126,17 +126,21 @@ jobs:
           Keep the installation's original binkw32.dll and mss32.dll; the build-time
           stub DLLs do not provide retail video or audio playback.
           "@ | Set-Content artifacts/INSTALL.txt
+          Compress-Archive -Path artifacts/* -DestinationPath "${{ matrix.game.artifact_name }}"
+          Compress-Archive -Path ci-artifacts/* -DestinationPath "${{ matrix.game.ci_artifact_name }}"
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply .github/workflows/build-windows.yml:130]: This is a false positive. `actions/download-artifact@v8` defaults to `skip-decompress: false`, which automatically decompresses zip artifacts upon download into `${{ runner.temp }}/game-artifact`.

The runtime files (`GeneralsXZH.exe`, `zlib1.dll`, `binkw32.dll`, `mss32.dll`) are extracted directly into the target directory without an enclosing nested folder. In CI run 34920340518 (job 104227880002), `Replay Tests (windows)` executed `Locate game runtime`, located all required binaries directly in `$env:RUNNER_TEMP/game-artifact`, and completed with 100% success.

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -69,61 +63,27 @@ jobs:
     uses: ./.github/workflows/build-linux-flatpak.yml
     with:
       preset: linux64-deploy
-      checkout_ref: refs/tags/${{ inputs.release_version }}
+      checkout_ref: ${{ inputs.release_mode == 'Dry Run' && github.sha || format('refs/tags/{0}', needs.create-tag.outputs.release_version) }}
     secrets: inherit
```
[the remark this answers is not in the document]
Comment (fbraz3) [reply .github/workflows/release.yml:67]: `secrets: inherit` is the repository-standard pattern across all reusable build workflow callers within this repository, matching `ci.yml` (lines 83, 92, 99). The callees (`build-linux-flatpak.yml`, `build-macos.yml`, `build-windows.yml`) are internal to the same repository and exclusively read the six `NGMP_*` repository secrets. Retaining `secrets: inherit` ensures parity between CI and Release workflows without divergence.

**Comment by fbraz3** [comment]
@coderabbitai Addressed in commit 0099347:
- In `scripts/build/linux/build-linux-flatpak.sh`, removed redundant creation and deletion of local `.ngmp-config.cmake`, writing exclusively to `cmake/ngmp_env.cmake` so developer-local overrides are never clobbered or removed.
- In `docs/WORKLOG/2026-09-DIARY.md`, removed the deployment-specific filesystem path `/data/docker/generalslogin/`.
